enumerable_method.rb

```diff
@@ -0,0 +1,124 @@
+public def my_each(arr)
+    for name in arr do
+        yield (name)
+    end
+end
```

Comment:
#my_each returns an Enumerator if no block is given
#my_each does not mutate the original array
#my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self
#my_each when a block is given when self is an array behaves like #each calls the given block once for each element in self
#my_each when a block is given when self is a range returns the range itself after calling the 
given block once for each element in self
#my_each when a block is given when self is a range behaves like #each calls the given block once for each element in self
#my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self